src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/DatepickerComponent.tsx

```diff
@@ -185,17 +188,22 @@ function DatepickerComponent({
     setValidDate(true); // we reset valid date => show error onBlur or when user is done typing
     setValidationMessages({});
     if (dateValue && dateValue.isValid()) {
-      const dateString =
+      const validDate = isValidDate(dateValue); // the date can have a valid format but not pass min/max validation
+      setDate(dateValue);
+      if (validDate) {
+        const dateString =
         timeStamp === true
           ? dateValue?.toISOString(true)
           : dateValue.format(DatePickerSaveFormatNoTimestamp);
-      setValidDate(isValidDate(dateValue)); // the date can have a valid format but not pass min/max validation
-      handleDataChange(dateString);
-      setDate(dateValue);
+        handleDataChange(dateString);
+      }
+
     } else if (!dateValue) {
       setDate(null);
       setValidDate(true);
```

Comment:
This can be removed to avoid another render. It is already set to `true` on line 185 (188 in new code), and as far as I can tell theres no other place it could be set to something else in between.

```suggestion
```

Reply:
Nice, good catch!